v2/tools/generator/internal/jsonast/swagger_type_extractor.go

```diff
@@ -287,26 +355,29 @@ func (extractor *SwaggerTypeExtractor) doesResponseRepresentARMResource(response
 	// the schema can either be directly included
 	if response.Schema != nil {
 
-		schema := MakeOpenAPISchema(
+		schema := *response.Schema
+		result := MakeOpenAPISchema(
+			nameFromRef(schema.Ref),
 			*response.Schema,
 			extractor.swaggerPath,
 			extractor.outputPackage,
 			extractor.idFactory,
 			extractor.cache)
 
-		return &schema, isMarkedAsARMResource(schema)
+		return &result, isMarkedAsARMResource(result)
 	}
 
 	// or it can be under a $ref
-	if response.Ref.GetURL() != nil {
-
-		refFilePath, refSchema, pkg := loadRefSchema(response.Ref, extractor.swaggerPath, extractor.cache)
+	ref := response.Ref
+	if ref.GetURL() != nil {
+		refFilePath, refSchema, packageAndSwagger := loadRefSchema(ref, extractor.swaggerPath, extractor.cache)
```

Comment:
minor: Should this (all over) be something like `packageAndPath`, given we're technically using a generic interface that doesn't _require_ Swagger?

Reply:
Unsure, since we really do require Swagger now.